lib/src/git.dart

```diff
@@ -121,13 +122,32 @@ String? repoRoot(String dir) {
   return null;
 }
 
+final _minSupportedGitVersion = Version(2, 14, 0);
```

Comment:
This needs a documentation string that explains what features were adding in Git 2.14 that causes it to the be the minimum supported version.

Reply:
We only used that in:
```sh
$ git grep recurse-submodules
lib/src/validator/gitignore.dart:        '--recurse-submodules'
```
So I'm guessing this is actually a problem we've caused, one could even call it a regression.

Reply:
This was used in Package.listFiles before that. So the regression is not introduced at that point.